internal/client/client.go

```diff
@@ -431,59 +447,121 @@ func (l *leveledLogger) Warn(msg string, keysAndValues ...interface{}) {
 
 func errorHandler(resp *http.Response, err error, numTries int) (*http.Response, error) {
 	var werr error
+	msg := fmt.Sprintf("request error after %d attempt(s)", numTries)
 	if err == nil {
-		err = errors.New(fmt.Sprintf("request error after %d attempt(s)", numTries))
+		err = errors.New(msg)
+	} else {
+		err = errors.Wrap(err, msg)
 	}
 
 	if resp != nil {
-		var orgid, reason, terrmsg, errmsg, retryAfter string
+		var orgid, reason, terrmsg, errmsg string
 		// TODO @mattdeekay: convert these to specific error types
 		if resp.Header != nil {
 			orgid = resp.Header.Get("X-Databricks-Org-Id")
 			reason = resp.Header.Get("X-Databricks-Reason-Phrase") // TODO note: shown on notebook
 			terrmsg = resp.Header.Get("X-Thriftserver-Error-Message")
 			errmsg = resp.Header.Get("x-databricks-error-or-redirect-message")
-			retryAfter = resp.Header.Get("Retry-After")
 			// TODO note: need to see if there's other headers
 		}
 		msg := fmt.Sprintf("orgId: %s, reason: %s, thriftErr: %s, err: %s", orgid, reason, terrmsg, errmsg)
```

Comment:
this error message is too cumbersome. As an user it would read strange. Better to keep the same logic as before.